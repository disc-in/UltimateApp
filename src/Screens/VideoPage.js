import React from 'react';
import { Platform, StyleSheet, View, Text, Share } from 'react-native';

import theme from '../styles/theme.style';
import I18n from '../utils/i18n';
import VimeoVideo from '../Components/shared/VimeoVideo';
import Button from '../Components/shared/Button';
import { showError } from '../utils/flashMessage';

const VideoPage = (props) => {
  const video = props.route.params.video;

  const share = () => {
    const url = video.youtube;

    Share.share({
      message: I18n.t('videoPage.shareContent', { url }),
      url,
    }).catch((e) => {
      showError(I18n.t('videoPage.error'));
    });
  };

  return (
    <View style={styles.videoPage}>
      <View style={styles.video}>
        <VimeoVideo vimeoId={video.video} sounds shouldPlay />
      </View>
      <View style={styles.title}>
        <Text style={styles.text}>{video.title}</Text>
        <Text style={styles.textAuthor}>{video.text}</Text>
      </View>
      {video.youtube && (
        <View style={styles.footer}>
          <Button
            onPress={share}
            icon={Platform.select({
              ios: 'ios-share-outline',
              default: 'share-social',
            })}
            text={I18n.t('videoPage.share')}
            testID="shareButton"
            small
            light
          />
        </View>
      )}
    </View>
  );
};

export default VideoPage;

const styles = StyleSheet.create({
  videoPage: {
    flex: 1,
    backgroundColor: theme.COLOR_PRIMARY_LIGHT,
  },
  video: {
    height: 250,
  },
  title: {
    marginVertical: 20,
    marginLeft: 10,
  },
  text: {
    color: theme.COLOR_PRIMARY,
    fontSize: theme.FONT_SIZE_MEDIUM,
    fontWeight: 'bold',
  },
  textAuthor: {
    color: theme.COLOR_SECONDARY,
    fontSize: theme.FONT_SIZE_SMALL,
  },
  footer: {
    position: 'absolute',
    paddingBottom: 20,
    paddingTop: 5,
    bottom: 0,
    width: '100%',
    alignItems: 'center',
  },
});
