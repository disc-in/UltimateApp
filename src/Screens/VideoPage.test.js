import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { NavigationContext } from '@react-navigation/native';
import { Share } from 'react-native';

import store from '../Store/testStore';

import VideoPage from './VideoPage';

jest.mock('../Components/shared/VimeoVideo');

describe('<VideoPage />', () => {
  const route = {
    params: {
      video: store.getState().theory.essentials[0].pages[0],
    },
  };
  // fake NavigationContext value data
  const navContext = {
    isFocused: () => true,
    // addListener returns an unscubscribe function.
    addListener: jest.fn(() => jest.fn()),
  };

  it('renders correctly', () => {
    const { toJSON } = render(
      <NavigationContext.Provider value={navContext}>
        <VideoPage route={route} />
      </NavigationContext.Provider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('triggers share', async () => {
    const share = jest.fn();
    Share.share = () => new Promise((resolve, reject) => share());

    const { getByTestId } = render(
      <NavigationContext.Provider value={navContext}>
        <VideoPage route={route} />
      </NavigationContext.Provider>,
    );

    await fireEvent.press(getByTestId('shareButton'));

    expect(share).toHaveBeenCalled();
  });
});
