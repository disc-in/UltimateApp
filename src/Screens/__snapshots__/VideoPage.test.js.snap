// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoPage /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": 250,
      }
    }
  >
    <View
      shouldPlay={true}
      sounds={true}
      testID="VimeoVideoMock"
      vimeoId="424000350"
    />
  </View>
  <View
    style={
      Object {
        "marginLeft": 10,
        "marginVertical": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#232b2b",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      How to make a perfect cut?
    </Text>
    <Text
      style={
        Object {
          "color": "#A6A6A6",
          "fontSize": 14,
        }
      }
    >
      Rise Up Ultimate
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
        "paddingBottom": 20,
        "paddingTop": 5,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#fff",
          "borderColor": "#2a9e91",
          "borderRadius": 5,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 30,
          "justifyContent": "center",
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 0,
          "width": "auto",
        }
      }
      testID="shareButton"
    >
      <Text />
      <Text
        style={
          Object {
            "color": "#2a9e91",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Share
      </Text>
    </View>
  </View>
</View>
`;
